common/environment/environment.cpp

```diff
@@ -31,11 +31,74 @@
 #include "dalienv.hpp"
 
 #define SDS_LOCK_TIMEOUT  30000
-
+#define DEFAULT_DROPZONE_INDEX      1
+#define DROPZONE_BY_MACHINE_SUFFIX  "-dropzoneByMachine-"
+#define DROPZONE_SUFFIX             "dropzone-"
+#define MACHINE_PREFIX              "machine-"
+#define NUMBER_OF_DROPZONES_SUFFIX  "-numberOfDropZones"
 
 static int environmentTraceLevel = 1;
 static Owned <IConstEnvironment> cache;
 
+class CLocalEnvironment;
+
+class CConstMachineInfoIterator : public  CSimpleInterfaceOf<IConstMachineInfoIterator>
+{
+public:
+    CConstMachineInfoIterator();
+
+    virtual bool first();
+    virtual bool next();
+    virtual bool isValid();
+    virtual IConstMachineInfo & query();
+    virtual unsigned count() const;
+
+protected:
+    IConstMachineInfo * curr = nullptr;
+    Owned<CLocalEnvironment> constEnv;
+    unsigned index = 1;
+    unsigned maxIndex = 0;
+};
+
+class CConstDropZoneInfoIterator : public CSimpleInterfaceOf<IConstDropZoneInfoIterator>
+{
+public:
+    CConstDropZoneInfoIterator();
+
+    virtual bool first();
```

Comment:
future: I've been liberally adding 'override' keywords in new code where appropriate, e.g. here
It's saved my bacon on more than one occasion already.